packages/flutter_tools/test/integration/test_driver.dart

```diff
@@ -128,6 +128,9 @@ class FlutterTestDriver {
       // expected by tests. Tests will reload/restart as required if they need
       // to hit breakpoints, etc.
       await waitForPause();
+      if (pauseOnExceptions) {
+        await (await getFlutterIsolate()).setExceptionPauseMode(VMExceptionPauseMode.unhandled);
```

Comment:
double await?
From the dart spec:

> We collapse multiple layers of futures into one. If e evaluates to a future f,
> the future will not invoke its then() callback until f completes to a non-future
> value, and so the result of an await is never a future, and the result of an async
> function will never have type Future< X > where X itself is an invocation of
> Future.

Reply:
I can't call `setExceptionPauseMode` until the first future completes, and the second await is for the result of that, so I think this is slightly different (this is `await (await a()).something()` not `await (await a())`. I didn't know about that behaviour though, it's good to know.

(If you think I can still improve this, lmk)